src/2d/shallow/surge/model_storm_module.f90

```diff
@@ -383,6 +383,92 @@ subroutine get_storm_data(t, storm, location, velocity, max_wind_radius,  &
     end subroutine get_storm_data
 
 
+    ! ==========================================================================
+    ! A set of common functions applied before calculating the circular component of 
+    ! wind speed using a particular wind model.
+ 
+    ! 1. Adjust max wind speed by removing translational speed (bounding at 0).
+
+    ! 2. Convert wind speed at 10 m to top of atmospheric boundary layer
+    ! ==========================================================================
+    subroutine adjust_max_wind(tv, mws, mod_mws)
+
+        real (kind=8), intent(inout) :: tv(2), mws
+
+        real (kind=8), intent(out) :: mod_mws
+
+        real (kind=8) :: trans_speed, trans_mod
+
+        trans_speed = sqrt(tv(1)**2 + tv(2)**2)
+        mod_mws = mws - trans_speed
+
+        ! Bound this at 0
+        if (mod_mws < 0) then
+            trans_mod = mws / trans_speed
+            tv = tv * trans_mod
+            mod_mws = 0
+        end if
+
+        ! Convert wind speed (10 m) to top of atmospheric boundary layer
+        mod_mws = mod_mws / atmos_boundary_layer
+
+    end subroutine adjust_max_wind
+
+    
+    ! ==========================================================================
+    ! A set of common functions applied after the circular component of wind speed
+    ! has been calculated given a particular wind model:
+ 
+    ! 1. Determine translation speed that should be added to final storm wind speed.
+    !    This is tapered to zero as the storm wind tapers to zero toward the eye of the 
+    !    storm and at long distances from the storm. Avoid divide-by-0 error
+
+    ! 2. Convert wind velocity from top of atmospheric boundary layer to wind
+    !    velocity at 10 m above the earth's surface.
+
+    ! 3. Convert from 1 minute averaged winds to 10 minute averaged winds
+
+    ! 4. Add back in translational velocity
+
+    ! 5. Apply distance ramp to limit scope
+    ! ==========================================================================
+    subroutine post_process_wind_estimate(maux, mbc, mx, my, i, j, wind, aux, &
```

Comment:
Again, is it possible to do with a `pure` function?

Reply:
Same thing here. I just slapped pure on there, since changing the subroutine to a function would be tricker since it modifies a couple of values.